packages/renderer/client/render.ts

```diff
@@ -23,41 +23,76 @@ export const render = async (
   try {
     const renderer = new Renderer(project);
 
-    const settings = {
-      ...project.meta.getFullRenderingSettings(),
-      name: project.name,
-    };
-
-    const firstFrame = renderer.timeToFrame(startInSeconds);
-
-    let frames: number;
-    console.log('startInSeconds', startInSeconds);
-    console.log('endInSeconds', endInSeconds);
-    console.log('!isFinite', !isFinite(endInSeconds));
-    console.log('startInSeconds === 0', startInSeconds === 0);
-    if (startInSeconds === 0 && !isFinite(endInSeconds)) {
-      console.log('option 1');
-      frames = await renderer.getNumberOfFrames(settings);
-    } else {
-      console.log('option 2');
-      frames = renderer.timeToFrame(endInSeconds) - firstFrame + 1;
-    }
-
-    // Get the range of frames to render (there can't be any overlap between workers)
-    const framesPerWorker = Math.ceil(frames / totalNumOfWorkers);
-    const startFrame = firstFrame + framesPerWorker * workerId;
-    const endFrame = Math.min(
-      startFrame + framesPerWorker,
-      startFrame + frames,
-    );
+    const {firstGlobalFrame, lastGlobalFrame} =
+      await getGlobalFirstAndLastFrame(
+        project,
+        renderer,
+        startInSeconds,
+        endInSeconds,
+      );
+    const {firstWorkerFrame, lastWorkerFrame} =
+      await getWorkerFirstAndLastFrame(
+        firstGlobalFrame,
+        lastGlobalFrame,
+        workerId,
+        totalNumOfWorkers,
+      );
 
     await renderer.render({
       ...project.meta.getFullRenderingSettings(),
       name: project.name,
-      range: [renderer.frameToTime(startFrame), renderer.frameToTime(endFrame)],
+      range: [
+        renderer.frameToTime(firstWorkerFrame),
+        renderer.frameToTime(lastWorkerFrame),
+      ],
     });
     window.onRenderComplete();
   } catch (e: any) {
     window.onRenderFailed(e.message);
   }
 };
+
+/**
+ * Calculate the first and last "global" frame, i.e. frame independent of worker
+ */
+async function getGlobalFirstAndLastFrame(
+  project: Project,
+  renderer: Renderer,
+  startSecondFromUser: number,
+  endSecondFromUser: number,
+) {
+  const firstGlobalFrame = renderer.timeToFrame(startSecondFromUser);
+  let lastGlobalFrame: number;
+
+  if (isFinite(endSecondFromUser)) {
+    // if end value is provided by user, lastFrame can be obtained
+    lastGlobalFrame = renderer.timeToFrame(endSecondFromUser);
+  } else {
+    // otherwise, endSecondFromUser is infinity, so lastGlobalFrame is the duration of the video
```

Comment:
This is ok to leave inside the else block i think. still needs to be capitalized though and also the period is missing